ui/component/or-scheduler/src/index.ts

```diff
@@ -0,0 +1,555 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+import {html, LitElement, PropertyValues, TemplateResult} from "lit";
+import {customElement, property, query, state} from "lit/decorators.js";
+import {CalendarEvent} from "@openremote/model";
+import {InputType, OrInputChangedEvent} from "@openremote/or-mwc-components/or-mwc-input";
+import {translate,i18next} from "@openremote/or-translate";
+
+import {OrMwcDialog, showDialog} from "@openremote/or-mwc-components/or-mwc-dialog";
+import {Frequency as FrequencyValue, RRule, Weekday, WeekdayStr} from 'rrule'
+import moment from "moment";
+import { Days } from "rrule/dist/esm/rrule";
+import { BY_RRULE_PARTS, EventTypes, FREQUENCIES, MONTHS, NOT_APPLICABLE_BY_RRULE_PARTS, rruleEnds } from "./data";
+import type { RulePartKey, RuleParts, LabeledEventTypes, Frequency } from "./types";
+import { when } from "lit/directives/when.js";
+export { RuleParts, RulePartKey, Frequency, LabeledEventTypes };
+
+export * from "./data";
+
+function range(start: number, end: number): number[] {
+    return Array.from({ length: end - start + 1 }, (_, i) => start + i )
+}
+
+// TODO: use es2023
+function toSpliced<T>(arr: T[], index: number, count: number): T[] {
+    arr.splice(index, count);
+    return arr;
+}
+
+const byWeekNoOptions = toSpliced(range(-53, 53), 53, 1);
+const byYearDayOptions = toSpliced(range(-366, 366), 366, 1); // TODO: optimize option rendering
+const byMonthDayOptions = toSpliced(range(-31, 31), 31, 1);
+const byHourOptions = range(1, 24);
+const byMinuteOrSecondsOptions = range(1, 60);
+
+export interface OrSchedulerChangedEventDetail {
+    value?: CalendarEvent;
+}
+
+export class OrSchedulerChangedEvent extends CustomEvent<OrSchedulerChangedEventDetail> {
+
+    public static readonly NAME = "or-scheduler-changed";
+
+    constructor(value?: CalendarEvent) {
+        super(OrSchedulerChangedEvent.NAME, {
+            detail: { value },
+            bubbles: true,
+            composed: true
+        });
+    }
+}
+
+declare global {
+    export interface HTMLElementEventMap {
+        [OrSchedulerChangedEvent.NAME]: OrSchedulerChangedEvent;
+    }
+}
+
+@customElement("or-scheduler")
+export class OrScheduler extends translate(i18next)(LitElement) {
+
+    @property()
+    public calendarEvent?: CalendarEvent = this.default;
+
+    @property()
+    public default?: CalendarEvent;
+
+    @property()
+    public disabledFrequencies: Frequency[] = [];
+
+    @property()
+    public disabledRRuleParts: RulePartKey[] = [];
+
+    @property()
+    public eventType: EventTypes = EventTypes.default;
+
+    @property()
+    public eventTypes: LabeledEventTypes = EventTypes;
+
+    @property()
+    public header?: string;
+
+    @property()
+    public isAllDay = true;
```

Comment:
Preferably these properties include a `@property({type: XXXX})`. It's very convenient in practice,
because `type: boolean` replaces `<or-scheduler isAllDay="true">` with `<or-scheduler isAllDay>`
It's also useful for documentation purposes.